ckanext/datastore/backend/postgres.py

```diff
@@ -1217,16 +1218,18 @@ def upsert_data(context: Context, data_dict: dict[str, Any]):
             rows.append(row)
 
         sql_string = '''INSERT INTO {res_id} ({columns})
-            VALUES ({values});'''.format(
+            VALUES ({values}) RETURNING {return_columns};'''.format(
             res_id=identifier(data_dict['resource_id']),
             columns=sql_columns,
+            return_columns=return_columns,
```

Comment:
we'd want to not return the values from the sql if they're going to be ignored (include_records is False)

Reply:
just to save on the extra sql work? yeah, makes sense

Reply:
the slow bit is probably parsing and looping over the values on the python side, but yeah

Reply:
@wardi as discussed, these actions will all default to `"include_records": False` and the sql and python loop for `records` will only happen if include_records is True